src/printer.js

```diff
@@ -1855,7 +1856,17 @@ function printStatementSequence(path, options, print) {
 
     parts.push(stmtPrinted);
 
-    if (util.isNextLineEmpty(text, stmt) && !isLastStatement(stmtPath)) {
+    if (parent.type === "SwitchCase") {
```

Comment:
This function is a generic one. Could you instead try to fix it inside of the implementation of SwitchCase?